File: pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float owe = 0;
    do
    {
        owe = get_float("Change owed: ");
    }
    while (!(owe > 0));

    int total = round(owe * 100);
    int coins[4] = {25, 10, 5, 1};
    int totalCoins = 0;

    for (int i = 0; i < 4; i++)
    {
        if (total < coins[i])
        {
            continue;
        }

        int mod = total % coins[i];
        int numberOfCoins = (total - mod) / coins[i];
        totalCoins += numberOfCoins;
        total = mod;
    }
    printf("%i\n", totalCoins);
}

File: pset6/cash/cash.py
from cs50 import get_float

owe = 0.0
while True:
    owe = get_float("Change owed: ")
    if owe > 0:
        break

total = round(owe * 100)
coins = [25, 10, 5, 1]
totalCoins = 0

for i in range(len(coins)):
    if total < coins[i]:
        continue

    mod = total % coins[i]
    numberOfCoins = (total - mod) / coins[i]
    totalCoins += numberOfCoins
    total = mod

print(f"{totalCoins:.0f}")